value-processor/src/org/immutables/value/processor/Criteria.generator

```diff
@@ -179,6 +203,19 @@ context.add(Expressions.path("[a.name]"))
 [/if]
 [/output.trim][/template]
 
+[template scalarMatcherType Attribute a][output.trim]
+[if a.unwrappedElementType eq 'boolean']
+  BooleanMatcher
+[else if a.unwrappedElementType eq 'java.lang.String']
+  StringMatcher
+[else if a.isMaybeComparableKey]
+  ComparableMatcher
+[else]
+  ObjectMatcher
+[/if]
+[/output.trim]
```

Comment:
this can give extra newline added, the one after `[/output.trim]` but before `[/template]`